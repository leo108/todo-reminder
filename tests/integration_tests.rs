use std::process::Command;

#[test]
fn test_main_with_table_format() {
    let output = Command::new("cargo")
        .arg("run")
        .arg("tests/test_files/config.toml")
        .arg("--format=table")
        .arg("--no-tty")
        .arg("--exit-zero")
        .output()
        .expect("Failed to execute command");

    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_debug_snapshot!(stdout.lines().collect::<Vec<&str>>());
}

#[test]
fn test_main_with_json_format() {
    let output = Command::new("cargo")
        .arg("run")
        .arg("tests/test_files/config.toml")
        .arg("--format=json")
        .arg("--exit-zero")
        .output()
        .expect("Failed to execute command");

    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    insta::assert_debug_snapshot!(stdout.lines().collect::<Vec<&str>>());
}
