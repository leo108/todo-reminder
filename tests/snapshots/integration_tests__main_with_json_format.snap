---
source: tests/integration_tests.rs
expression: "stdout.lines().collect::<Vec<&str>>()"
---
[
    "[",
    "  {",
    "    \"comment\": \"# TODO: incorrect format\",",
    "    \"file\": \"bash/test.sh\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"# TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"bash/test.sh\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"c/test.c\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"c/test.c\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"c/test.c\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"c/test.c\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"c_sharp/test.cs\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"c_sharp/test.cs\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"c_sharp/test.cs\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"c_sharp/test.cs\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"cpp/test.cpp\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"cpp/test.cpp\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"cpp/test.cpp\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"cpp/test.cpp\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/* TODO: incorrect format */\",",
    "    \"file\": \"css/test.css\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/* TODO: 2023-10-01 @alice.smith Implement the function */\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"css/test.css\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"css/test.css\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"css/test.css\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"go/test.go\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"go/test.go\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"go/test.go\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"go/test.go\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"java/test.java\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"java/test.java\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"java/test.java\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"java/test.java\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"javascript/test.js\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"javascript/test.js\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"javascript/test.js\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"javascript/test.js\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"php/test.php\",",
    "    \"line\": 4,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"php/test.php\",",
    "    \"line\": 5,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"php/test.php\",",
    "    \"line\": 10,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"php/test.php\",",
    "    \"line\": 13,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"\\\"\\\"\\\"\\nTODO fix some issues\\n\\\"\\\"\\\"\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 1,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"\\\"\\\"\\\"\\n    TODO: 2023-10-01 @alice.smith Implement the function\\n    \\\"\\\"\\\"\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 6,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"\\\"\\\"\\\"\\n        TODO: incorrect format\\n        \\\"\\\"\\\"\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 17,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"\\\"\\\"\\\"\\n    TODO: 2023-10-01 @alice.smith Implement the function\\n    \\\"\\\"\\\"\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 23,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"# TODO: incorrect format\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 27,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"# TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"python/test.py\",",
    "    \"line\": 28,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"# TODO: incorrect format\",",
    "    \"file\": \"ruby/test.rb\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"# TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"ruby/test.rb\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"=begin\\nTODO: incorrect format\\n=end\",",
    "    \"file\": \"ruby/test.rb\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"=begin\\nTODO: 2023-10-01 @alice.smith Implement the function\\n=end\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"ruby/test.rb\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"rust/no_due.rs\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"rust/no_due.rs\",",
    "    \"line\": 7,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"rust/no_format.rs\",",
    "    \"line\": 2,",
    "    \"owner\": \"alice\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"rust/no_format.rs\",",
    "    \"line\": 7,",
    "    \"owner\": \"alice\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"rust/test.rs\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"rust/test.rs\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"rust/test.rs\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"rust/test.rs\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: incorrect format\",",
    "    \"file\": \"typescript/test.ts\",",
    "    \"line\": 2,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"// TODO: 2023-10-01 @alice.smith Implement the function\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"typescript/test.ts\",",
    "    \"line\": 3,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: incorrect format\\n*/\",",
    "    \"file\": \"typescript/test.ts\",",
    "    \"line\": 8,",
    "    \"type\": \"InvalidFormat\"",
    "  },",
    "  {",
    "    \"comment\": \"/*\\nTODO: 2023-10-01 @alice.smith Implement the function\\n*/\",",
    "    \"due_date\": \"2023-10-01\",",
    "    \"file\": \"typescript/test.ts\",",
    "    \"line\": 11,",
    "    \"owner\": \"alice.smith\",",
    "    \"type\": \"Overdue\"",
    "  }",
    "]",
]
